### PR TITLE
#410 inbound detail view 2

### PR DESCRIPTION
**Pull request by joshxg**
More of #410

- This one moves around a bunch of types.
- Hopefully it's a good idea - I moved all the invoice related types from common to the `invoices` package. That's the only place they're used. If they're going to be used elsewhere my thinking is that it's likely that it would be a different query and might need a different type anyway, rather than messing with these ones 🤷 
- Additionally made the `Invoice`/`InvoiceRow` types extend the gql `InvoiceNode`
- I also noticed that the `operations.graphql` file is going to be ridiculous. Same with the `schema.graphql`. I'm thinking that at least the `operations.graphql` file should live next to the `api.ts` file in each submodule which those operations are for.. though there will likely be a bunch of duplicates... probably. But, I do think we need to partition this file somehow as we edit it manually. `schema.graphql` is a copy paste of the graphql schema file from the api which is not ideal but at least somewhat automated.. i'd really like to split that out also, but it would make the process very tedious and annoying. Any ideas?

### CODE REVIEW

File: packages/common/src/operations.graphql [hunk 1]
```diff
@@ -27,6 +27,7 @@ query invoice($id: String!) {
       shippedDatetime
       deliveredDatetime
       enteredByName
+      donorName
```
Comment (joshxg) [inline packages/common/src/operations.graphql:30]: Ah I just realised this was my addition to the type, but the generated code was melting down

File: packages/common/src/types/index.ts [hunk 2]
```diff
@@ -52,78 +51,3 @@ export type Store = {
   id: string;
   name: string;
 };
-
-export interface InvoiceLine extends InvoiceLineNode, DomainObject {
```
Comment (joshxg) [inline packages/common/src/types/index.ts:56]: So all of this moved to `invoices` package.. Think the other ones should go in their packages also - i.e. items/name into system. Stock lines.. uhh.. system also?

File: packages/common/src/ui/components/domain/StatusCrumbs/StatusCrumbs.stories.tsx [hunk 3]
```diff
@@ -1,36 +1,30 @@
...
 import Radio from '@mui/material/Radio';
 import RadioGroup from '@mui/material/RadioGroup';
 import FormControlLabel from '@mui/material/FormControlLabel';
 import FormControl from '@mui/material/FormControl';
 import FormLabel from '@mui/material/FormLabel';
+import { StatusCrumbs } from './StatusCrumbs';
+import { LocaleKey, useTranslation } from '../../../../intl';
 
-const StatusTranslation: Record<OutboundShipmentStatus, LocaleKey> = {
+const statusTranslation: Record<string, LocaleKey> = {
```
Comment (joshxg) [inline packages/common/src/ui/components/domain/StatusCrumbs/StatusCrumbs.stories.tsx:12]: did some status changes a little further down so just changed this story up a bit..

File: packages/invoices/src/InboundShipment/ListView/ListView.tsx [hunk 4]
```diff
@@ -48,13 +48,11 @@ export const InboundListView: FC = () => {
...
-    onUpdate({ ...row, color: color.hex });
-  };
-
-  const columns = useColumns<InvoiceRow>(
+  const columns = useColumns(
     [
-      getNameAndColorColumn(onColorUpdate),
+      getNameAndColorColumn((row: InvoiceRow, color: Color) => {
+        onUpdate({ ...row, color: color.hex });
+      }),
```
Comment (joshxg) [inline packages/invoices/src/InboundShipment/ListView/ListView.tsx:55]: not sure why 🤔

File: packages/invoices/src/InboundShipment/ListView/api.ts [hunk 5]
```diff
@@ -56,7 +55,7 @@ const getSortKey = (sortBy: SortBy<Invoice>): InvoiceSortFieldInput => {
   }
 };
 
-const getSortDesc = (sortBy: SortBy<Invoice>): boolean => {
+const getSortDesc = (sortBy: SortBy<InvoiceRow>): boolean => {
```
Comment (joshxg) [inline packages/invoices/src/InboundShipment/ListView/api.ts:58]: this is the list view, so uses invoice row not invoice!

File: packages/invoices/src/OutboundShipment/DetailView/Footer.tsx [hunk 6]
```diff
@@ -52,79 +52,81 @@ export const Footer: FC<OutboundDetailFooterProps> = ({ draft, save }) => {
...
-            }}
-            labelKey="label.hold"
-          />
-
-          <StatusCrumbs
-            statuses={outboundStatuses}
-            statusLog={createStatusLog(draft)}
-            statusFormatter={getStatusTranslation}
-          />
+        draft && (
```
Comment (joshxg) [inline packages/invoices/src/OutboundShipment/DetailView/Footer.tsx:55]: hmm I guess I dislike the optional chaining

Comment (joshxg) [reply packages/invoices/src/OutboundShipment/DetailView/Footer.tsx:55]: nothing else really changed here - just the calls for statuses as the draft is now definitely a draft.. 🙄

File: packages/invoices/src/OutboundShipment/DetailView/index.ts [hunk 7]
```diff
@@ -1,38 +1 @@
-import { OutboundShipment } from '../../types';
 export { DetailView } from './DetailView';
-
-export const placeholderInvoice: OutboundShipment = {
```
Comment (joshxg) [inline packages/invoices/src/OutboundShipment/DetailView/index.ts:4]: put this in `invoices/utils`. Though, now that I think about it. Why?

File: packages/common/src/types/index.ts (hunk 2 above)
Comment (mark-prins) [reply packages/common/src/types/index.ts:56]: hmm, yeah. stock lines are shared across a number of areas.. might be best to have them in system

File: packages/invoices/src/OutboundShipment/DetailView/Footer.tsx (hunk 6 above)
Comment (mark-prins) [reply packages/invoices/src/OutboundShipment/DetailView/Footer.tsx:55]: fair enough! that was a lot of chaining

File: packages/invoices/src/OutboundShipment/DetailView/index.ts (hunk 7 above)
Comment (mark-prins) [reply packages/invoices/src/OutboundShipment/DetailView/index.ts:4]: either place is fine - it is a utils kind of thing 🤷

File: packages/invoices/src/types.ts [hunk 8]
```diff
@@ -86,3 +134,45 @@ export type OutboundShipmentSummaryItem = {
...
+  itemId: string;
+  itemCode: string;
+  itemName: string;
+  unitQuantity: number;
+  numberOfPacks: number;
+  locationDescription?: string | null;
+  itemUnit?: string;
+  batch?: string | null;
+  batches: Record<string, OutboundShipmentRow>;
+  sellPrice?: number | undefined;
```
Comment (mark-prins) [inline packages/invoices/src/types.ts:166]: I was wondering what the differences are between In & Out BoundShipments, and here's one! Outbound will show `sellPrice` and Inbound should show `costPrice`.

All the same, both types can have both props - it's only the column list which needs to differ!

**Review by mark-prins — APPROVED** [review]
🎉  Nice! very sensible moving of stuff around generally.

I agree about the `operations.graphql` getting too large! not sure how to handle the splitting of the `schema.graphql` though.. if we are taking it directly from the API then it would be easiest to simply copy, regardless of size.. though I am worried about how big it will get.

File: packages/invoices/src/types.ts (hunk 8 above)
Comment (joshxg) [reply packages/invoices/src/types.ts:166]: Yeah there's really not much differences right now and maybe not ever, maybe it's backwards but if it turns out there's no difference we can consolidate

**Pull request merged**